### DISCUSSION TITLE
Scheduled TimeWarp Program

### DISCUSSION THREAD

**Discussion by knowenR** [gopro/labs#1192] [Q&A]
Hello
i want to create a daily timewarp program that it will run between 08:00 - 19:00. if 
current time > 19:00 gopro must be ready for the next day 08:00

code is running between the specified hours but "else command" is  not running. What can be wrong 

My QR Code -->  !MBOOT="!Lbt"!SAVEbt=>08:00<19:00!SmTWpAr4fL
0dR30+!R~!08:00NQ

Activated Features: !MTUSB=1, !MWAKE=2, !MBOOT

![Ekran görüntüsü 2024-12-02 093711](https://github.com/user-attachments/assets/1744535e-f3d6-429c-a068-9bb4f5826231)

**Comment by dnewman-gpsw** [comment]
There are a number issue here, and one Labs bug.

The added QR command can't be used that way within the nest, and the command is wrong (starting, then settings.)

Currently there is a bug with >08:00<19:00 commands that use else, fixed in the next release.  But you will not need that structure for this project.

Make it simple:  !08:00NmTW!S!19:00E!2R

There maybe other functions you were intending, but start simple.

**Comment by knowenR** [comment]
thanks for answer

I want to install this device on tugboat. and For 20 days tugboat will travel to other countries.  no one to take care of the device during this time.  go pro has to do everything itself.

I will supply the device with electricity continuouslyI.  if there is a power outage the program has to resume where it left off.
it must be only run between 08:00 - 19:00. if there is a problem during this time interval, such as a power outage, the program should continue to run (if it is in the time interval, it should not start at 08:00 the next day)

**Comment by dnewman-gpsw** [comment]
Then test your test should be ```<08:00!08:00R>19:00!08:00RmTW!S!19:00E!2R```  <- this will do the trick

**Comment by knowenR** [comment]
My GoPro is HeroBlack 10
when i scan this code. It takes a timewarp shot counting down from 30 seconds and then stops.

**Reply by knowenR** [reply]
thanks david following test code worked for me. 

!MBOOT="!Lbt"!SAVEbt=>08:00<13:43mTW+r1+fW+!
S+!30E+!1OR!07:00NQ+!R

**Reply by dnewman-gpsw** [reply]
You had the duration set to 30s in your TimeWarp preset, turn that off and my script is more reliable. There is no good reason to have settings in every loop (they add failure points.) Just set up the preset once (for TimeWarp) and use a simpler script. Your first script had dR30 (duration 30s) and your latest has !30E, another 30s limit. Is there another requirement for hundreds of 30s timewarps?

Be warned only HERO12 and 13 are robust enough to run for 20 days without human intervention. HERO10 may do it if you keep the script simple and only booting once per day.
